task/10-katas-1-tasks.js

```diff
@@ -56,7 +88,27 @@ function createCompassPoints() {
  *   'nothing to do' => 'nothing to do'
  */
 function* expandBraces(str) {
-    throw new Error('Not implemented');
+    const queue = [],
+        regex = new RegExp("\{([0-9a-zA-Z\.,]+)\}", 'i'),
+        itemSet = new Set;
+
+    queue.push(str);
+
+    while (queue.length) {
+        let item = queue.shift();
+        let matches = item.match(regex);
+
+        if (matches != null) {
```

Comment:
"operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
'0' != 0
and '0' !== 0)"